test/sd59x18/gm/gm.t.sol

```diff
@@ -0,0 +1,131 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity >=0.8.13;
+
+import {
+    E,
+    MAX_SD59x18,
+    MAX_WHOLE_SD59x18,
+    MIN_SD59x18,
+    MIN_WHOLE_SD59x18,
+    PI,
+    PRBMathSD59x18__GmOverflow,
+    PRBMathSD59x18__GmNegativeProduct,
+    SD59x18,
+    ZERO,
+    gm
+} from "~/SD59x18.sol";
+import { SD59x18__BaseTest } from "../SD59x18BaseTest.t.sol";
+
+contract SD59x18__GmTest is SD59x18__BaseTest {
+    // Biggest number whose square fits within int256
+    SD59x18 internal constant SQRT_MAX_SD59x18_DIV_BY_SCALE = SD59x18.wrap(240615969168004511545_033772477625056927);
+    // Smallest number whose square fits within int256
+    SD59x18 internal constant SQRT_MIN_SD59x18_DIV_BY_SCALE = SD59x18.wrap(-240615969168004511545_033772477625056927);
+
+    function zeroSets() internal returns (Set[] memory) {
+        delete sets;
+        sets.push(set({ x: ZERO, y: PI, expected: ZERO }));
+        sets.push(set({ x: PI, y: ZERO, expected: ZERO }));
+        return sets;
+    }
+
+    function testGm__OneOperandZero() external parameterizedTest(zeroSets()) {
+        SD59x18 actual = gm(s.x, s.y);
+        assertEq(actual, s.expected);
+    }
+
+    modifier NotZeroOperands() {
+        _;
+    }
+
+    function negativeSets() internal returns (Set[] memory) {
+        delete sets;
+        sets.push(set({ x: -7.1e18, y: 20.05e18, expected: ZERO }));
+        sets.push(set({ x: -1e18, y: PI, expected: ZERO }));
+        sets.push(set({ x: PI, y: -1e18, expected: ZERO }));
+        sets.push(set({ x: 7.1e18, y: -20.05e18, expected: ZERO }));
```

Comment:
No need to even pass the `expected` argument here. There is an overload of the `set` function which takes only `x` and `y` as arguments.